src/main/scala/is/hail/expr/AST.scala

```diff
@@ -691,6 +724,25 @@ case class Apply(posn: Position, fn: String, args: Array[AST]) extends AST(posn,
     case (_, _) =>
       FunctionRegistry.call(fn, args, args.map(_.`type`).toSeq)
   }
+
+  private val tryPrimOpConversion: PartialFunction[IndexedSeq[IR], IR] = {
+    val pf: PartialFunction[IndexedSeq[IR], Option[IR]] = {
```

Comment:
It's also annoying pattern match blocks can't infer their types.  We could define:

```
def lift[T, S](pf: PartialFunction[T, S]): (T) => S = pf.lift
def flatLift[T, S](pf: PartialFunction[T, Option[S]]): (T) => Option[S] = pf.flatLift // assuming name change
```

Then we can write:

```
  flatLift[IndexedSeq[IR], IR] {
    case IndexedSeq(x) => ...
  }
```

This feels more direct since in every case you define a partial function you immediately lift it.

Reply:
Mostly just playing with code organization ideas, don't feel like this is binding.